pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -531,12 +532,12 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 					updated = true
 					// add the pod podSetUpdates
 					checkState.PodSetUpdates = podSetUpdates(wl, pr)
+					// propagate the message from the provisioning request status into the workload
+					// to change to the "successfully provisioned" message after provisioning
 					updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned).Message)
 				}
-			case prAccepted:
-				// we propagate the message from the provisioning request status into the workload
-				// this happens for provisioned = false (ETA updates) and also for provisioned = true
-				// to change to the "successfully provisioned" message after provisioning
+			case prAccepted && prProvisionedFalse:
+				// propagate the ETA update from the provisioning request into the workload
 				updated = updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned).Message) || updated
 				updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
```

Comment:
```suggestion
			case prAccepted:
				if provisionedCond := apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned); provisionedCond != nil {
					// propagate the ETA update from the provisioning request into the workload
					updated = updateCheckMessage(&checkState, provisionedCond.Message) || updated
					updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
				}
```

It seems not to matter if the `Provisined` condition is False since the upon switch case `case prProvisioned:` already checked it, right?